### ISSUE TITLE
[Bug] VZV | Total number of fatalities does not match fatalities in mode for 2022

### ISSUE THREAD

**Issue by patrickm02L** [cityofaustin/atd-data-tech#9245]
The Total number of fatalities for 2022 is correct as shown in the fatalities widget in the VZV. The Fatalities in travel mode do not add up to total fatalities - There are 36 instead of 38. Joel M. pointed out that Motorist should have an additional fatality and Pedestrian should include an additional fatalities.

![Screen Shot 2022-05-23 at 3.03.11 PM.png](https://images.zenhubusercontent.com/5d68810b05d7792664be403f/5cd128a2-581a-488a-8ac6-9549f3c31192)

Wondering if this is related to an earlier issue #8129.

**Comment by patrickm02L** [comment]
@joelmeyer21 please add anything I've missed.

**Comment by joelmeyer21** [comment]
> @joelmeyer21 please add anything I've missed.

Correct. However, we do not know if the missing Pedestrian is a pedestrian or a scooter (i.e. we're not sure if you are classifying scooters as Pedestrians or Other in VZV). Thanks.

**Comment by mateoclarke** [comment]
@joelmeyer21 Do you have the crash ID for the incident involving a Scooter fatality?

**Comment by joelmeyer21** [comment]
> @joelmeyer21 Do you have the crash ID for the incident involving a Scooter fatality?

221190173

**Comment by Charlie-Henry** [comment]
Pulled the Socrata dataset's crashes by year and mode: None the totals are matching the `apd_confirmed_death_count`.

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:////Users/charliehenry/Library/Group%20Containers/UBF8T346G9.Office/TemporaryItems/msohtmlclip/clip.htm">
<link rel=File-List
href="file:////Users/charliehenry/Library/Group%20Containers/UBF8T346G9.Office/TemporaryItems/msohtmlclip/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">
<meta charset=utf-8>



crash_date | motor_vehicle_death_count | bicycle_death_count | motorcycle_death_count | pedestrian_death_count | other_death_count | Sum of modes | apd_confirmed_death_count
-- | -- | -- | -- | -- | -- | -- | --
2022 | 11 | 2 | 10 | 14 | 0 | 37 | 38
2021 | 56 | 3 | 14 | 43 | 1 | 117 | 114
2020 | 53 | 4 | 6 | 35 | 0 | 98 | 92
2019 | 39 | 4 | 12 | 35 | 1 | 91 | 88
2018 | 32 | 2 | 9 | 30 | 0 | 73 | 74
2017 | 40 | 4 | 11 | 23 | 0 | 78 | 75
2016 | 35 | 2 | 13 | 29 | 0 | 79 | 77
2015 | 53 | 3 | 15 | 31 | 0 | 102 | 101
2014 | 34 | 0 | 13 | 12 | 0 | 59 | 56
2013 | 41 | 1 | 12 | 21 | 0 | 75 | 75
2012 | 25 | 2 | 10 | 12 | 0 | 49 | 49



</body>

</html>

**Comment by Charlie-Henry** [comment]
Nevermind, we are actually using `death_cnt` as the field for that summary card for fatals, updated the table I've attached.

**Comment by patrickm02L** [comment]
@xavierapostal

**Comment by patrickm02L** [comment]
@joelmeyer21 We talked about how to approach solving the data inconsistencies in an afternoon developer's meeting, and have some pretty good leads.  

@mateoclarke is going to take a first pass to see if we can identify the record(s) that need to be updated or if there's a more comprehensive solution that we'll need to take. We're looking to have more answers by tomorrow 5/24/22, and are aware that there is upcoming media that references the VZV. 

Has the media been published yet or do you know when it will be published?

**Comment by joelmeyer21** [comment]
> @joelmeyer21 We talked about how to approach solving the data inconsistencies in an afternoon developer's meeting, and have some pretty good leads.
> 
> @mateoclarke is going to take a first pass to see if we can identify the record(s) that need to be updated or if there's a more comprehensive solution that we'll need to take. We're looking to have more answers by tomorrow 5/24/22, and are aware that there is upcoming media that references the VZV.
> 
> Has the media been published yet or do you know when it will be published?

We don't know if there will be any news stories, but the crash stats were published in ATD's newsletter yesterday (which goes to media outlets and the public) so we want to make sure those stats align with the VZV.

**Comment by joelmeyer21** [comment]
Just wanted to check in to see if you all have been able to identify the source of the discrepancies yet. Thanks.

**Comment by mateoclarke** [comment]
I spent some time yesterday (5/30) manually looking through the YTD fatality records. At that point in time, there was only 1 fatality discrepancy between the top line fatality count and the sum of fatalities by mode. This is different from the 2 count difference when this was initial reported. (I'm not sure why the discrepancy shrank from 2 to 1 between last week and yesterday).

After manually checking most of the 40 crashes with fatalities, I found that the [Crash record with ID 18865794](https://visionzero.austin.gov/editor/#/crashes/18865794) had a discrepancy.

Before my update the record had this value for `atd_mode_category_metadata`:
```json
[
    {"mode_id": 1, "mode_desc": "Passenger car", "unit_id": 4651994, "death_cnt": 0, "sus_serious_injry_cnt": 0, "nonincap_injry_cnt": 0, "poss_injry_cnt": 0, "non_injry_cnt": 0, "unkn_injry_cnt": 0, "tot_injry_cnt": 0}, 
    {"mode_id": 2, "mode_desc": "Large passenger vehicle", "unit_id": 4651995, "death_cnt": 0, "sus_serious_injry_cnt": 0, "nonincap_injry_cnt": 0, "poss_injry_cnt": 1, "non_injry_cnt": 0, "unkn_injry_cnt": 0, "tot_injry_cnt": 1}, 
    {"mode_id": 1, "mode_desc": "Passenger car", "unit_id": 4651997, "death_cnt": 0, "sus_serious_injry_cnt": 1, "nonincap_injry_cnt": 0, "poss_injry_cnt": 0, "non_injry_cnt": 0, "unkn_injry_cnt": 0, "tot_injry_cnt": 1}
]
```
The important thing to note is that `death_cnt` for each of the 3 vehicles involved was 0 even though this crash had `atd_fatality_count` = 1.

Updated the mode metadata for this crash, changing unit ID 4651997 from `sus_serious_injry_cnt` from 1 to 0 and `death_cnt` from 0 to 1. The data was correct in VZE and in the unit & person tables in VZD.

![Screen Shot 2022-05-31 at 12 53 41 PM](https://user-images.githubusercontent.com/5697474/171268467-aaca3eb1-0f9c-4a13-a5f3-1e5918608f49.png)

One thing that is notable is that this crash has a 0 death count according to CRIS. I believe the reason this discrepancy happened is because the `atd_mode_category_metadata` field is populated whenever the Crash Record is updated. But it does not get triggered to recalculate when a related Unit or Person record changes. A relatively small change to our back end DB logic would be to automatically update `crash.atd_mode_category_metadata` anytime the unit record updates.

[Here is the trigger function](https://github.com/cityofaustin/atd-vz-data/blob/e99a1ab0bb1f0bd7e4b45bb0f8ebad93c4a42b37/atd-vzd/triggers/get_crash_modes.sql) that calculates `atd_mode_category_metadata`. [Here is where that function is invoked](https://github.com/cityofaustin/atd-vz-data/blob/c1d6a920f6aba15bbea51ac32d863f9f7f8bf0a9/atd-vzd/triggers/atd_txdot_crashes_updates_audit_log.sql#L152) as an automatic trigger on a crash record update.
